packages/jmix-react-core/src/crud/validation.ts

```diff
@@ -0,0 +1,209 @@
+                    const valSeconds = value.hour() * 60 * 60 + value.minute() * 60 + value.second();
+
+                    return valSeconds <= nowSeconds;
+                }
+                default: throw new Error('Not implemented');
+            }
+        }
+        case 'Future': {
+            if (isNull(value)) {
+                return true;
+            }
+
+            switch (dataType) {
+                case 'DateTime':
+                case 'LocalDateTime':
+                case 'OffsetDateTime':
+                    return dayjs(value).isAfter(new Date());
+                case 'Date':
+                case 'LocalDate': {
+                    return dayjs(value).isAfter(new Date(), 'date');
+                }
+                case 'Time':
+                case 'LocalTime':
+                case 'OffsetTime': {
+                    const now = dayjs();
+                    const nowSeconds = now.hour() * 60 * 60 + now.minute() * 60 + now.second();
+                    const valSeconds = value.hour() * 60 * 60 + value.minute() * 60 + value.second();
+
+                    return valSeconds > nowSeconds;
+                }
+                default: throw new Error('Not implemented');
+            }
+        }
+        case 'FutureOrPresent': {
+            if (isNull(value)) {
+                return true;
+            }
+
+            switch (dataType) {
+                case 'DateTime':
+                case 'LocalDateTime':
+                case 'OffsetDateTime':
+                    return dayjs(value).isAfter(dayjs()) || dayjs(value).isSame(dayjs());
+                case 'Date':
+                case 'LocalDate': {
+                    return dayjs(value).isAfter(dayjs(), 'date') || dayjs(value).isSame(dayjs(), 'date');
+                }
+                case 'Time':
+                case 'LocalTime':
+                case 'OffsetTime': {
+                    const now = dayjs();
+                    const nowSeconds = now.hour() * 60 * 60 + now.minute() * 60 + now.second();
+                    const valSeconds = value.hour() * 60 * 60 + value.minute() * 60 + value.second();
+
+                    return valSeconds >= nowSeconds;
+                }
+                default: throw new Error('Not implemented');
+            }
+        }
+        case 'Pattern': {
+            if (isNull(value)) {
+                return true;
+            }
+
+            const matches = String(value).match(new RegExp(rule.regexp, rule.modifiers));
+
+            return matches?.length === 1 && matches[0].length === String(value).length;
+        }
+        case 'Size': {
+            return isNull(value) || (typeof value === 'string' || Array.isArray(value)) && value.length >= rule.min && value.length <= rule.max;
+        }
+        default: throw new Error('Not implemented');
+    }
+}
+
+function isNull(value: any) {
+    return value === null || typeof value === 'undefined' || value === '';
+}
```

Comment:
Some parts of this code looks imperatively and bulky. Maybe it will be better to bring it out to helper functions. @vyacheslav-pushkin What do you think?